### ISSUE TITLE
Suspense fallback attribute should not contain JSX

### ISSUE THREAD

**Issue by minecrawler** [oxc-project/oxc#3242]
Hi oxc team!

I'm unsure about the specifics, since I'm still learning React, so I want to report this as a mismatch with React docs.

According to the [React docs](https://react.dev/reference/react/Suspense), the following is valid:

```tsx
const Component = () => {
  return <Suspense fallback={<span>Loading...</span>}>...</Suspense>;
};
```

However oxlint tells me that "JSX attribute values should not contain other JSX.", which sounds like I should change it to this:

```tsx
const Component = () => {
  const loadingEleRef = useRef<ReactNode>(null);

  if (loadingEleRef.current === null) {
    loadingEleRef.current = <span>Loading...</span>;
  }

  return <Suspense fallback={loadingEleRef.current}>...</Suspense>;
};
```

What's the correct way?

**Comment by Dunqing** [comment]
Both ways are correct. But the `jsx-no-jsx-as-prop` rule thought second way performance is better. Since you are a React beginner, You just follow React docs.

You can use `-A` to turn off this rule.
```shell
oxlint -A jsx-no-jsx-as-prop
```
